### PR TITLE
Alternative preminted tokenIds

### PR DESCRIPTION
**Pull request by zajck**
Draft PR to see the necessary changes to change tokenId management

### CODE REVIEW

File: contracts/protocol/clients/voucher/BosonVoucher.sol [hunk 1]
```diff
@@ -772,42 +777,28 @@ contract BosonVoucherBase is
...
-                            committable = true;
-                            offerId = currentOfferId;
-                            owner = range.owner;
-                        }
-                        break; // Found!
-                    } else {
-                        // No? It may be in a higher range
-                        low = mid + 1;
-                    }
+            // it might be a pre-minted token. Preminted tokens have offerId in the upper 128 bits
```
Comment (mischat) [inline contracts/protocol/clients/voucher/BosonVoucher.sol:782]: Does the indentation here look wrong ?

File: contracts/protocol/clients/voucher/BosonVoucher.sol [hunk 2]
```diff
@@ -327,6 +328,12 @@ contract BosonVoucherBase is
             end = start + maxPremintedVouchers;
         }
 
+        // Update last burned token id
+        range.lastBurnedTokenId = end - 1;
+
+        start += (_offerId << 128);
```
[the remark this answers is not in the document]
Comment (zajck) [reply contracts/protocol/clients/voucher/BosonVoucher.sol:334]: I guess we can do a shift already in `reserveRange` and avoid all other shifts. 
I will test it.

File: contracts/protocol/clients/voucher/BosonVoucher.sol (hunk 1 above)
Comment (zajck) [reply contracts/protocol/clients/voucher/BosonVoucher.sol:782]: Hmm, prettier did it this way.

**Comment by zajck** [comment]
Replaced by #583

**Pull request closed without merging**